Fix incorrect symlink path when using relative dir

tweska opened the pull request:
I ran into a problem when I put a project in the `bundle.yml` and use the `dir` keyword to point at a relative project directory:
- Relative to the `bundle.yml` will create a broken symlink;
- Relative to the source directory will not even create a symlink the first time, but will create the correct symlink the second time (I assume this is not intended behavior...);
- Setting the absolute path will always work, but requires to put the absolute path or `${path}` in the `bundle.yml`.

This fix will convert a relative path to the project directory into an absolute path before calling the symlink function. If a relative path is used, the path should be relative to the `bundle.yml`.

Additionally replaced `os.path` with `path` (there was both `import os` and `from os import path`), and I made the CI happy. :^)

FussyDuck commented:
[![CLA assistant check](https://bol-claassistant.ecmwf.int:443/pull/badge/not_signed)](https://bol-claassistant.ecmwf.int:443/ecmwf/ecbundle?pullRequest=26) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf/ecbundle?pullRequest=26) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf/ecbundle?pullRequest=26) it.</sub>

wdeconinck commented:
Hi @tweska I changed the target branch for the PR to develop, which now also contains PR #23 . There are now merge conflicts to be resolved.

tweska commented:
I rebased and updated the PR. To create the symlink, we will now choose the path in this order:
1. absolute path
2. relative to the bundle file
3. relative to the source directory

wdeconinck reviewed (approved):
That looks great! Thanks @tweska ! 🙏
